common2/gratia/common2/timeutil.py

```diff
@@ -87,24 +89,24 @@ def strfdelta(tsec, format_str="P", format_no_day="PT{H}H{M}M{S}S", format_zero=
         f = Formatter()
         d = {}
         l = {'D': 86400, 'H': 3600, 'M': 60, 'S': 1}
-        rem = long(tsec)
+        rem = int(tsec)
 
         if format_str == "P":
             # variable format
             if 0 < tsec < 86400:
                 format_str = "PT"
             for i in ('D', 'H', 'M', 'S'):
-                if i in l.keys():
+                if i in list(l.keys()):
                     d[i], rem = divmod(rem, l[i])
                     if d[i] != 0:
                         format_str = "%s{%s}%s" % (format_str, i, i)
         else:
             if 0 < tsec < 86400:
                 format_str = format_no_day
-            k = map(lambda x: x[1], list(f.parse(format_str)))
+            k = [x[1] for x in list(f.parse(format_str))]
```

Comment:
```suggestion
            k = [x[1] for x in f.parse(format_str)]
```
Since you're iterating over it, you don't need to convert it to a list first.

Reply:
That is true, but i left it in because that stylistic oddity existed in the original  :)

Nevertheless i can't otherwise justify keeping it.